File: src/backend/contratos/src/model/contrato_historico_reajuste_model.py
from src import db


class ContratoHistoricoReajusteModel(db.Model):
    __tablename__ = 'contrato_historico_reajuste'

    id = db.Column(db.Integer, primary_key=True)
    indice = db.Column(db.Float)
    valor_anterior = db.Column(db.Float)
    valor_atual = db.Column(db.Float)
    data_reajuste = db.Column(db.DateTime)
    observacao = db.Column(db.Text)
    id_contrato = db.Column(db.Integer, db.ForeignKey('contrato.id'))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_contrato = data.get('idContrato')
        self.indice = data.get('indice')
        self.valor_anterior = data.get('valorAnterior')
        self.valor_atual = data.get('valorAtual')
        self.data_reajuste = data.get('dataReajuste')
        self.observacao = data.get('observacao')

    def serialize(self):
        return {
            'id': self.id,
            'idContrato': self.id_contrato,
            'indice': self.indice,
            'valorAnterior': self.valor_anterior,
            'valorAtual': self.valor_atual,
            'dataReajuste': self.data_reajuste.isoformat(),
            'observacao': self.observacao,
        }

File: frontend/contratos/lib/app/data/model/contrato_historico_reajuste_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:contratos/app/infra/infra_imports.dart';
import 'package:intl/intl.dart';

class ContratoHistoricoReajusteModel {
	int? id;
	int? idContrato;
	double? indice;
	double? valorAnterior;
	double? valorAtual;
	DateTime? dataReajuste;
	String? observacao;

	ContratoHistoricoReajusteModel({
		this.id,
		this.idContrato,
		this.indice,
		this.valorAnterior,
		this.valorAtual,
		this.dataReajuste,
		this.observacao,
	});

	static List<String> dbColumns = <String>[
		'id',
		'indice',
		'valor_anterior',
		'valor_atual',
		'data_reajuste',
		'observacao',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Indice',
		'Valor Anterior',
		'Valor Atual',
		'Data Reajuste',
		'Observacao',
	];

	ContratoHistoricoReajusteModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idContrato = jsonData['idContrato'];
		indice = jsonData['indice']?.toDouble();
		valorAnterior = jsonData['valorAnterior']?.toDouble();
		valorAtual = jsonData['valorAtual']?.toDouble();
		dataReajuste = jsonData['dataReajuste'] != null ? DateTime.tryParse(jsonData['dataReajuste']) : null;
		observacao = jsonData['observacao'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idContrato'] = idContrato != 0 ? idContrato : null;
		jsonData['indice'] = indice;
		jsonData['valorAnterior'] = valorAnterior;
		jsonData['valorAtual'] = valorAtual;
		jsonData['dataReajuste'] = dataReajuste != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataReajuste!) : null;
		jsonData['observacao'] = observacao;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idContrato = plutoRow.cells['idContrato']?.value;
		indice = plutoRow.cells['indice']?.value?.toDouble();
		valorAnterior = plutoRow.cells['valorAnterior']?.value?.toDouble();
		valorAtual = plutoRow.cells['valorAtual']?.value?.toDouble();
		dataReajuste = Util.stringToDate(plutoRow.cells['dataReajuste']?.value);
		observacao = plutoRow.cells['observacao']?.value;
	}	

	ContratoHistoricoReajusteModel clone() {
		return ContratoHistoricoReajusteModel(
			id: id,
			idContrato: idContrato,
			indice: indice,
			valorAnterior: valorAnterior,
			valorAtual: valorAtual,
			dataReajuste: dataReajuste,
			observacao: observacao,
		);			
	}

	
}